libraries/resource_providers/api500/c7000/volume_provider.rb

```diff
@@ -20,33 +20,29 @@ def load_resource_with_associated_resources
           else
             validate_required_properties(:storage_system, :storage_pool)
             storage_system_uri = load_storage_system
-            @item['storagePoolUri'] = load_resource(:StoragePool, { name: @new_resource.storage_pool, storageSystemUri: storage_system_uri }, 'uri')
-            @item['snapshotPoolUri'] = load_resource(:StoragePool, { name: @new_resource.snapshot_pool, storageSystemUri: storage_system_uri }, 'uri') if @new_resource.snapshot_pool
+            @item.set_storage_pool(resource_named(:StoragePool).new(@item.client, name: @new_resource.storage_pool, storageSystemUri: storage_system_uri))
+            @item.set_snapshot_pool(resource_named(:StoragePool).new(@item.client, name: @new_resource.snapshot_pool, storageSystemUri: storage_system_uri)) if @new_resource.snapshot_pool
           end
-          set_properties unless @item.exists?
+          @item['properties']['name'] = @new_resource.name
```

Comment:
As ```set_storage_pool``` and ```set_snapshot_snapshot``` are called, the ```properties``` attribute is present in the data because in the API500, the ```name``` property is inside the ```properties```. In the ```exists?``` method in the SDK will search by name within ```properties```. If you do not think this is cool, I'll have to improve the methods in the SDK as describe [here](https://github.com/HewlettPackard/oneview-sdk-ruby/issues/282).